src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.fruits.containsKey(transaction.getFruit())
+                && transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+            int newValue = Storage.fruits.get(transaction.getFruit())
+                    - transaction.getQuantity();
+            if (newValue < 0) {
+                throw new RuntimeException("Result can't be negative number");

```

Comment:
The RuntimeException is a very general exception. It would be better to use a more specific exception, or even a custom exception.

Reply:
ok, done